src/main/cpp/domconfigurator.cpp

```diff
@@ -331,10 +321,12 @@ void DOMConfigurator::parseErrorHandler(Pool& p,
 	AppenderMap& appenders)
 {
 
-	ErrorHandlerPtr eh = OptionConverter::instantiateByClassName(
+	ErrorHandlerPtr eh;
+	std::shared_ptr<Object> obj = OptionConverter::instantiateByClassName(
```

Comment:
There are multiple places using `std::shared_ptr` like above, shouldn't that be `log4cxx::` instead?

Reply:
[Only at least C++11 will be supported](https://lists.apache.org/thread.html/re744feb89b0bf9a15b03d145a88fe0d0be48ad069d12a31b8ba5d461%40%3Cdev.logging.apache.org%3E) in future, so it doesn't make too much sense anymore to migrate those calls to `log4cxx::`. Instead, the project is simply going to depend on [std::shared_ptr without BOOST-fallback](https://lists.apache.org/thread.html/r80a4dbbb719684c899223a1540e3d608c8fa74928dd9adfe46a5ae06%40%3Cdev.logging.apache.org%3E), so the above code can either stay as-is or might be changed to use already available macros `LOG4CXX_PTR_DEF`.